src/main/java/proj/kedabra/billsnap/presentation/controllers/BillController.java

```diff
@@ -109,15 +108,15 @@ public BillSplitResource getDetailedBill(
         return billMapper.toResource(detailedBill);
     }
 
+    @CachePut(value = CacheNames.BILL, key = "#associateBillResource.id + #principal.name")
+    @CacheEvict(value = CacheNames.PAYMENTS, key = "#principal.name")
```

Comment:
why do you evict for payments all the time?

Reply:
So anytime we do modifications on a bill right, adding items or paying items or whatever, your payments owed should also change, so that the next time you call for a payment owed, it has to be updated to reflect the new changes. Else, the data cached won't be in line with the updated bills

Reply:
You can merge it if that definition makes sense